src/components/Footer.tsx

```diff
@@ -0,0 +1,37 @@
+import { useState, useEffect } from 'react';
+
+export default function Footer() {
+	const [time, setTime] = useState<Date>(new Date());
+	const [lastUpdate, setLastUpdate] = useState<Date>(new Date());
+
+	useEffect(() => {
+		setInterval(() => setTime(new Date()), 1000);
+		setInterval(() => setLastUpdate(new Date()), import.meta.env.VITE_MAP_UPDATE_INTERVAL as number);
```

Comment:
Implementation looks good, however it would make more sense to me if LastUpdate time would be updated on sucessfully receiving an update from the API. I this case the API could not be live and thus no data could be provided, but the frontend could still say it is updating properly.

This however would require a large system overhaul so dont bother with the limited time we have.

Reply:
I did try to do this, but it triggered infinite rerenders of the component (also creating a infinite amount of setIntervals, leading to A LOT of memory usage), so i decided on this approach. i agree that it is bad but all i could do right now